include/picongpu/fields/incidentField/Functors.hpp

```diff
@@ -469,7 +469,8 @@ namespace picongpu
                     template<typename T_SeparableFunctor>
                     HDINLINE float3_X operator()(T_SeparableFunctor const& functor, floatD_X const& totalCellIdx) const
                     {
-                        auto const time = functor.getCurrentTime(totalCellIdx);
+                        auto const time
+                            = functor.getCurrentTime(totalCellIdx) - T_SeparableFunctor::Unitless::TIME_DELAY;

```

Comment:
IMO this change should be done in the implementation of `getCurrentTime()` directly.
The reason for this is that now all places where `getCurrentTime()` is used we substitute the delay and this is an easy source to introduce errors by forgetting places.

Please update the implementation of `getCurrentTime()` and update the doxygen string with a hint that the time can be negative.

Reply:
The time delay is moved into `getCurrentTime()` now